src/components/authorize/authorize-controller.ts

```diff
@@ -32,12 +38,27 @@ function createConsentCookie(
 
 export function authorizeGet(
   authService: AuthorizeServiceInterface = authorizeService(),
-  cookieService: CookieConsentServiceInterface = cookieConsentService()
+  cookieService: CookieConsentServiceInterface = cookieConsentService(),
+  kmsService: KmsDecryptionServiceInterface = new KmsDecryptionService(),
+  jwtService: JwtServiceInterface = new JwtService()
 ): ExpressRouteFunc {
   return async function (req: Request, res: Response) {
     const { sessionId, clientSessionId, persistentSessionId } = res.locals;
     const loginPrompt = sanitize(req.query.prompt as string);
 
+    if (req.query.request !== undefined) {
+      const encryptedAuthRequestJWE = req.query.request as string;
+      const authRequestJweDecryptedAsJwt = await kmsService.decrypt(
+        encryptedAuthRequestJWE
+      );
+
+      const claims = await jwtService.getPayloadWithSigCheck(
+        authRequestJweDecryptedAsJwt
+      );
+
+      jwtService.validateClaims(claims);
+    }
```

Comment:
We should probably throw when it is undefined

Reply:
validateClaims always either returns claims or throws an error. This will be changed once we know what were doing with the claims.

Reply:
if statement was unnecessary - removed so change in comment not needed.